src/core/src/lib/utils.ts

```diff
@@ -118,6 +118,10 @@ export function clone(value: any): any {
     return value.slice(0).map(v => clone(v));
   }
 
+  if (value.clone && value.clone()) {
```

Comment:
`clone` method seems like a workaround, the right solution is to find a better way to clone a class instance.
could try the following solution instead:
```ts
// https://stackoverflow.com/questions/41474986/how-to-clone-a-javascript-es6-class-instance
value = Object.assign(Object.create( Object.getPrototypeOf(value)), value);
```

Reply:
+ add some tests :)